uo_sys_framework/ACREModule/cfgFunctions/common/fn_addRack.sqf

```diff
@@ -0,0 +1,148 @@
+		case "ACRE_VRC111": {_mountedRadio = "ACRE_PRC148";};
+		case "ACRE_VRC64": {_mountedRadio = "ACRE_PRC77";};
+		default {
+			(format ["(ACRE_SETUP_MODULE) [ERR]: NFB ERROR ENCOUNTERED"]) call UO_FW_FNC_DebugMessage;
+			(format ["(ACRE_SETUP_MODULE) [ERR]: VERIFY YOUR RACK CLASSNAME: %1", _rackType]) call UO_FW_FNC_DebugMessage;
+		};
+	};
+} else {_mountedRadio = "";};
+
+_args pushBackUnique _mountedRadio;
+
+if(isNil "_rackNameLong" || isNil "_rackNameShort") then {
+	(format ["(ACRE_SETUP_MODULE) [ERR]: A RACKNAME IS NIL, LONG: %1 SHORT: %2", _rackNameLong, _rackNameShort]) call UO_FW_FNC_DebugMessage;
+	(format ["(ACRE_SETUP_MODULE) [ERR]: DEFAULTING TO LONGNAME: ""DEFAULT RACK NAME: FIX ME"" AND SHORTNAME: ""FIX"""]) call UO_FW_FNC_DebugMessage;
+	_rackNameLong = "DEFAULT RACK NAME: FIX ME";
+	_rackNameShort = "FIX";
+};
+
+_args pushBackUnique _rackNameLong;
+_args pushBackUnique _rackNameShort;
+
+if(_whitelistedPositions isEqualTo [] && _blacklistedPositions isEqualTo []) then {
+	(format ["(ACRE_SETUP_MODULE) [ERR]: NO POSITIONS HAVE BEEN WHITELISTED OR BLACKLISTED"]) call UO_FW_FNC_DebugMessage;
+	(format ["(ACRE_SETUP_MODULE) [ERR]: DEFAULTING TO 'inside' and 'all'"]) call UO_FW_FNC_DebugMessage;
+	_whitelistedPositions = ["inside", "all"];
+} else {
+	if(_whitelistedPositions isEqualTo _blacklistedPositions) then {
+		(format ["(ACRE_SETUP_MODULE) [ERR]: ALL POSITIONS ARE WHITELISTED AND BLACKLISTED"]) call UO_FW_FNC_DebugMessage;
+		(format ["(ACRE_SETUP_MODULE) [ERR]: DEFAULTING WHITELISTED POSITIONS TO: 'inside'"]) call UO_FW_FNC_DebugMessage;
+		_whitelistedPositions = ["inside"];
+		_blacklistedPositions = [];
+	};
+};
+
+_args pushBackUnique _whitelistedPositions;
+_args pushBackUnique _blacklistedPositions;
+
+_args pushBackUnique _faction;
+
+_statement = {
+	if(!(isNil "_faction") && {UO_FW_ACRE_SCRAMBLE_ENABLED}) then {
+		private _faction_i = 3;
+		switch (_faction) do {
+			case west: {_faction_i = 0};
+			case east: {_faction_i = 1};
+			case independent: {_faction_i = 2};
+			case civilian: {_faction_i = 3};
+			default {_faction_i = 3;};
+		};
+		private _preset = _radioScramblePresets select _faction_i;
+		[_object, _preset] call acre_api_fnc_setVehicleRacksPreset;
+	};
+
+	[_object, {}] call acre_api_fnc_initVehicleRacks;
+	[_object, [_rackType, _rackNameLong, _rackNameShort, _radioRemovable, _whitelistedPositions, _blacklistedPositions, _mountedRadio, [], []], false, {}] call acre_api_fnc_addRackToVehicle;
+	if(UO_FW_DEBUG_ENABLED) then {(format ["ACRE_SETUP: RACK ADDED: %1", _rackNameLong]) call UO_FW_FNC_DebugMessage;};
+};
+
+_timeout = 30;
+
+_timeoutCode = {
+	(format ["(ACRE_SETUP_MODULE) [ERR]: ERROR ENCOUNTERED ACRE WAS NOT INITALIZED AFTER 30 SECONDS, UO_FW_FNC_addRack TIMED OUT"]) call UO_FW_FNC_DebugMessage;
+	(format ["(ACRE_SETUP_MODULE) [ERR]: ISSUE MAY BE RELATED TO LOCALHOSTING"]) call UO_FW_FNC_DebugMessage;
+};
+
+if ([] call acre_api_fnc_isInitialized) then {
```

Comment:
Shouldn't this be in the condition? And since `acre_api_fnc_isInitialized` is never set until CBA_missionTime is > 0 I think it's safe to put this as the only condition.

Reply:
Good catch. I am revising this to be one of the conditions including CBA_missionTime as a condition just in case. It's better to be safe rather than sorry.